### ISSUE TITLE
[Accessibility], WCAG 4.1.2. The semantics of the tree items on the Hierarchy and Groups tabs are inadequate

### ISSUE THREAD

**Issue by Eiskore** [NatLibFi/Skosmos#1981]
## Description of the problem
There is a Hierachy and Groups tabs on vocabularies. Both tab panels contains expendable tree pattern. All the elements in a tree view are semantically links. However, the link is not the correct role in the tree view, because the page does not actually change, and the focus does not move.  Screen reader users also now do not receive any feedback on the activation of the link, or whether the activation has done anything at all.

There are also focusable arrow buttons on tree view.  The screen reader finds the arrow-buttons and reads the name “Open button”. However, the screen reader does not tell whether the structure is open or closed. When you click the button (and the tree view closes or opens), the screen reader is completely silent.

**FIX**: To make the structure and choice of term is clearer with a screen reader, the concept list should be implemented as an ARIA treeview (role="tree" with role="treeitem"), not as a list of links. Selecting a term updates content within the same page but does not change context, so links are semantically incorrect. Expand/collapse arrows shouldn’t be separate buttons; they should be purely visual indicators of state, with expansion controlled via aria-expanded and keyboard arrow keys. 
Tree item selection could be indicated with aria-selected, and changes in the main content area announced using an aria-live region if focus remains in the tree.

Keyboard interaction must follow the ARIA tree pattern: Up/Down Arrow moves between visible items, Right Arrow expands a collapsed item, Left Arrow collapses an expanded item, Enter or Space selects the current item, and Home/End moves to the first or last item. 

More information about tree pattern: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/treeitem_role 

Severity: moderate
Priority: medium

## Additional information (e.g. screenshots) about the problem

<img width="1004" height="518" alt="Image" src="https://github.com/user-attachments/assets/223a13f3-8b7d-40d8-88d9-3083d44021e4" />

## The browser you used when the problem appeared
Chrome, Firefox, Edge
